### PR TITLE
rust: avoid infinite recursion in chain building

### PR DESCRIPTION
**Pull request by tetsuo-cpp**

### CODE REVIEW

**Comment by tetsuo-cpp** [comment]
This is the cause of the segfault we were seeing before. I think the reason why `load_der_public_key` was coming up in the stack trace was because `Policy.public_key` gets called in the chain building.

File: src/rust/cryptography-x509-validation/src/lib.rs [hunk 1]
```diff
@@ -77,6 +77,7 @@ where
         // * Search by AKI and other identifiers?
         self.intermediates
             .iter()
+            .filter(move |candidate| candidate != &cert)
```
Comment (tetsuo-cpp) [inline src/rust/cryptography-x509-validation/src/lib.rs]: I need to read up a bit and figure out why I need the `move` here. But I think this does the trick.

Comment (woodruffw) [reply src/rust/cryptography-x509-validation/src/lib.rs]: Yep, the logic here looks fine. We should probably fold it into the subsequent `filter(...)` for concision though, I'll look into that now.

**Review by woodruffw — APPROVED** [review]
LGTM! @tetsuo-cpp, feel free to merge if this looks reasonable to you as well 🙂

**Pull request merged**